### PR TITLE
update(panel): Adds absolute positioning with center.

### LINKED ISSUE #7794: mdPanel: Add centering for absolute positioning.
I have the perfect screensize, so this has always worked for me!

### PR DESCRIPTION
**Pull request by ErinCoughlan**
Fixes #7794, #7877 

@jelbourn @KarenParker @gmoothart @DerekLouie Please review.

### CODE REVIEW

File: src/components/panel/panel.js [hunk 1]
```diff
@@ -335,7 +335,7 @@ angular
  * Overlapping the panel with an element:
  * `new MdPanelPosition()
  *     .relativeTo(someElement)
- *     .withPanelXPosition('align-left')
+ *     .withPanelXPosition('align-start')
```
Comment (KarenParker) [inline src/components/panel/panel.js]: Consider having these options defined and accessible for the user, similar to the $mdPanel.animation values.

**Comment by KarenParker** [comment]
LGTM

File: src/components/panel/panel.js (hunk 1 above)
Comment (jelbourn) [inline src/components/panel/panel.js]: I agree that they should be available "enum-style"

File: src/components/panel/panel.js [hunk 2]
```diff
@@ -1487,58 +1550,95 @@ MdPanelPosition.prototype._calculatePanelPosition = function(panelEl) {
...
-      this._left = targetRight + 'px';
-      break;
-  }
+/**
+ * Calculates the panel position for absolute based on the created
+ * panel element.
+ * @param {number} panelWidth
+ * @param {number} panelHeight
+ */
+MdPanelPosition.prototype._calculateAbsolutePosition =
```
Comment (jelbourn) [inline src/components/panel/panel.js]: Line limit should be 100, so I think this will fit.

File: src/components/panel/panel.js [hunk 3]
```diff
@@ -1487,58 +1550,95 @@ MdPanelPosition.prototype._calculatePanelPosition = function(panelEl) {
...
+    function(panelWidth, panelHeight) {
+      var windowWidth = window.innerWidth;
```
Comment (jelbourn) [inline src/components/panel/panel.js]: You can center something completely with CSS:
```css
.globally-centered {
  top: 50%;
  left: 50%;
  transform: translateX(-50%) translateY(-50%);
}
```

Comment (jelbourn) [inline src/components/panel/panel.js]: Here's what I currently have for the Angular 2 version (subject to change as we alpha):
https://github.com/angular/material2/blob/master/src/core/overlay/position/global-position-strategy.ts

File: src/components/panel/panel.js (hunk 2 above)
Comment (ErinCoughlan) [inline src/components/panel/panel.js]: I think this file has been using 80, so I'd rather stay consistent in the file. I'm fine if we want to reflow everything to 100, but not in this PR.

**Comment by ErinCoughlan** [comment]
PTAL. I switched to using the transform css instead of calculating position (which is great because now I don't need to recalculate on page resize!)

File: src/components/panel/panel.js [hunk 4]
```diff
@@ -1051,6 +1094,7 @@ MdPanelRef.prototype._configurePosition = function() {
   this._panelEl.css('bottom', positionConfig.getBottom(this._panelEl));
   this._panelEl.css('left', positionConfig.getLeft(this._panelEl));
   this._panelEl.css('right', positionConfig.getRight(this._panelEl));
+  this._panelEl.css('transform', positionConfig.getTransform());
```
Comment (jelbourn) [inline src/components/panel/panel.js]: One thing to keep in mind is that an element can only have one `transform` value, so setting it directly will override any transforms used for animations.

**Comment by jelbourn** [comment]
LGTM

**Comment by KarenParker** [comment]
LGTM

**Pull request merged**